src/types/types.rs

```diff
@@ -0,0 +1,61 @@
+use ff::PrimeField;
+use poseidon_rs::{Fr};
+
+use crate::prover::sparse_merkle_tree::{BranchKey, BranchNode};
+use crate::prover::errors::MerkleTreeError;
+
+const ZERO_U32: u32 = 0;
+
+/// Fr traits
+pub trait ToFr {
+    fn to_fr(&self) -> Fr;
+}
+
+/// Convert a u32 into a field element.
+impl ToFr for u32 {
+    fn to_fr(&self) -> Fr {
+        Fr::from_str(&self.to_string()).unwrap()
+    }
+}
+
+pub trait ToVecFr {
+    fn to_vec_fr(&self) -> Vec<Fr>;
+}
+
+/// Convert a u32 into a vector of field element.
+impl ToVecFr for u32 {
+    fn to_vec_fr(&self) -> Vec<Fr> {
+        vec![Fr::from_str(&self.to_string()).unwrap()]
```

Comment:
Haha, there must be a better way to get a `u32` value than serializing it to string and then parsing it back. This is kinda like copying and pasting text by making a screenshot :) Very heavy machinery used for something that seems to me like it should be a very lightweight operation.

Reply:
Also, how big is the internal representation of the field elements? Can we really assume that they will always fit in 32 bits?